internal/idutils/bucketname.go

```diff
@@ -18,12 +18,51 @@ package idutils
 
 import (
 	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/dynatrace/dynatrace-configuration-as-code/v2/internal/featureflags"
+	"github.com/dynatrace/dynatrace-configuration-as-code/v2/internal/log"
 	"github.com/dynatrace/dynatrace-configuration-as-code/v2/pkg/config/coordinate"
 )
 
 // GenerateBucketName returns the "bucketName" identifier for a bucket based on the coordinate.
 // As all buckets are of the same type and never overlap with configs of different types on the same API, the "type" is omitted.
 // Since the bucket API does not support colons, we concatenate them using underscores.
+// If SanitizeBucketNames feature flag is enabled, the name will be sanizited to something matching `([a-z])([a-z0-9])([a-z0-9_-])+` with a max length of 100.
 func GenerateBucketName(c coordinate.Coordinate) string {
-	return fmt.Sprintf("%s_%s", c.Project, c.ConfigId)
+	name := fmt.Sprintf("%s_%s", c.Project, c.ConfigId)
+
+	if !featureflags.SanitizeBucketNames.Enabled() {
+		return name
+	}
+
+	sanitiziedName := SanitizeBucketName(name)
+	if sanitiziedName != name {
+		log.Warn("Bucket name was changed to '%s' from '%s'", sanitiziedName, name)
+	}
+
+	return sanitiziedName
+}
+
+// SanitizeBucketName modifies the specified name to meet the requirements of the bucket-definitions API: pattern: `([a-z])([a-z0-9])([a-z0-9_-])+`,  maxLength: 100.
+// It does this by deleting invalid characters and truncating the result if it is more than 100 characters long.
+func SanitizeBucketName(name string) string {
```

Comment:
Is this method necessary to be public? I would either 1) make the test file use the same package, or 2) only test the GenerateBucketName, but with all test cases that you already provide.

Reply:
Exactly that - I wanted to test just this function but the existing tests are in the `idutils_test` package.

Reply:
Thanks, I have fixed the tests now to use `GenerateBucketName` directly.